File: packages/3/ms-3-flaskpy/flaskr/routes/v1/estate_routes.py
from gmahechas_erp_common_flaskpy import Blueprint
from gmahechas_erp_common_flaskpy import Blueprint, request, Response, authentication_middleware

estate_blueprint = Blueprint('estate', __name__, url_prefix='/estate')


@estate_blueprint.route('/create/one', methods=['POST'])
@authentication_middleware
def create_one():
    return 'create/one'


@estate_blueprint.route('/update/one', methods=['PUT'])
@authentication_middleware
def update_one():
    return 'update/one'


@estate_blueprint.route('/delete/one', methods=['DELETE'])
@authentication_middleware
def delete_one():
    return 'delete/one'


@estate_blueprint.route('/search/one', methods=['POST'])
@authentication_middleware
def search_one():
    return 'search/one'


@estate_blueprint.route('/search/many', methods=['POST'])
@authentication_middleware
def search_many():
    return 'search/many'

File: packages/3/ms-3-expressjs/src/routes/v1/estate.routes.ts
import { iresponse, createOneEstateSchema, updateOneEstateSchema, deleteOneEstateSchema, searchOneEstateSchema, searchManyEstateSchema } from '@gmahechas/erp-common-js';
import { express, authenticationMiddleware, validatorMiddleware, loggerMiddleware } from '@gmahechas/erp-common-expressjs';
import { createOneEstate, updateOneEstate, deleteOneEstate, searchOneEstate, searchManyEstate } from '@gmahechas/erp-common-ms-3-js';

const router = express.Router();

router.post('/create/one', authenticationMiddleware, validatorMiddleware(createOneEstateSchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await createOneEstate(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.put('/update/one', authenticationMiddleware, validatorMiddleware(updateOneEstateSchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await updateOneEstate(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.delete('/delete/one', authenticationMiddleware, validatorMiddleware(deleteOneEstateSchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await deleteOneEstate(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.post('/search/one', authenticationMiddleware, validatorMiddleware(searchOneEstateSchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await searchOneEstate(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

router.post('/search/many', authenticationMiddleware, validatorMiddleware(searchManyEstateSchema), loggerMiddleware, async (request: express.Request, response: express.Response, next: express.NextFunction) => {
	const data = await searchManyEstate(request.body);
	const { statusCode, body } = iresponse(200, data);
	response.status(statusCode).send(body);
});

export const estateRouter = router;